cmd/scip/tests/snapshots/output/forward-def/forward_def.repro

```diff
@@ -0,0 +1,8 @@
+ reference forward_definition abc#
```

Comment:
nit: I would expect the repro syntax to work like this
```suggestion
forward_definition abc#
```

Reply:
I put it inside `reference` so that it's clearer that it is also a reference, not a tri-state of definition/reference/forward definition.

Reply:
Should we update the docstring to emphasize that you should set `Definition` and `ForwardDefinition`? It might be easy to misunderstand that setting `ForwardDefinition` alone implies `Definition`

Reply:
No, that would be incorrect. You should NOT set `Definition` if you set `ForwardDefinition` because a `ForwardDefinition` is actually a reference, it is not a Definition because it doesn't have a body.

I've added a lint for this in the `scip lint` command and mentioned the above requirement in the doc comment.

Reply:
It is a definition IMO because it's a valid destination for "Go to definition". When I trigger "Go to implementation" then I would expect it to go to the non-forward definition.

Reply:
My 2c based on working in C++: for Go to Definition, I always want to be taken to the method body if available, not to the forward declaration. Yes, if the Definition isn't available, then a forward declaration is OK as a fallback.

The "implementation" terminology is a bit confusing here, because with Find Impls, there are different possible Impls for the same interface & there is some kind of subtyping involved. However, with forward declarations, there should only be one place where the body of the method exists, multiple can't exist, so lumping the body of the method along with corresponding methods in subclasses (in C++, doesn't apply to C) would be confusing.

Reply:
If you still disagree, let's discuss it synchronously some time next week. It would be helpful for me to show some live examples of code nav behavior in C++.

Reply:
Or updating the docstring to emphasize that `Definition` should not be set

Reply:
Ugh, I thought I updated the docstring earlier, but I must've stashed that change instead of merging it. 🤦🏽 

Yes, I'll do that.

Reply:
In my mental model, it is a definition with lower priority. If you have two ambiguous definitions and one has the `ForwardDefinition` role then we can discard it. I would rather use that application logic than requiring the client to treat a non-definition (which is confusingly named `ForwardDefinition`) as a definition